core/storage/suggestion/gae_models.py

```diff
@@ -2428,28 +2428,24 @@ def fetch_page(
 
         sorted_results: List[
             TranslationSubmitterTotalContributionStatsModel] = []
-        today = datetime.date.today()
-
-        if max_days_since_last_activity is not None:
-            last_date = today - datetime.timedelta(
-                days=max_days_since_last_activity)
-            next_offset = offset
-            while len(sorted_results) < page_size:
-                result_models: Sequence[
-                    TranslationSubmitterTotalContributionStatsModel] = (
-                    sort_query.fetch(
-                        NUM_MODELS_PER_FETCH, offset=next_offset))
-                if not result_models:
+
+        next_offset = offset
+        while len(sorted_results) < page_size:
```

Comment:
I see three almost identical pieces of code the only difference here is type. Please extract it to a function

Reply:
This still stands, please address

Reply:
Done